### DISCUSSION TITLE
Bodies seem to repel each other even though no collision in config

### DISCUSSION THREAD

**Discussion by Theo-Cheynel** [google/brax#285] [General]
Hi,

I have a config file defining a humanoid ragdoll. I have removed all of the collisions inside this config. I expected the bodies to be able to intersect each other, instead, here's a video of their behaviour :

https://user-images.githubusercontent.com/34627497/210821915-f13a7056-6712-4e31-89cd-2f8eaf14e030.mp4

I am just applying a predefined series of actions at each step and observing the result, and we can see that they behave like they're repulsing each other so as not to collide, despite the collisions being turned off.

Does anyone know what is happening ? If necessary I can provide the config used and the script used to generate that result, as well as the data.

**Comment by btaba** [comment]
Hi @Theo-Cheynel ! It looks to me like the capsules are interpenetrating without a collision update. To confirm, can you check system.colliders to make sure it's empty?
https://github.com/google/brax/blob/main/brax/physics/config.proto#L103-L104

**Reply by Theo-Cheynel** [reply]
Here is the content of system.colliders :
```
[<brax.physics.colliders.TwoWayCollider at 0x7fdd04e8f6d0>,
 <brax.physics.colliders.OneWayCollider at 0x7fdd04e8fa60>,
 <brax.physics.colliders.TwoWayCollider at 0x7fdd04e8fdc0>,
 <brax.physics.colliders.TwoWayCollider at 0x7fdd04e8f850>]
```

Please find attached the config file I used, if ever you want to inspect it to understand where these colliders could be coming from. I only have bodies, joints and actuators in that config so I have no idea where the collisions are coming from. It's not a big deal for what I'm trying to do since I want the collisions anyway, but I wanted to remove them for visualization purposes at first to make sure I was giving the right actions.

**Reply by Theo-Cheynel** [reply]
[amp_ragdoll_no_floor.zip](https://github.com/google/brax/files/10411911/amp_ragdoll_no_floor.zip)
(providing it as a ZIP because for some reason github doesn't like the file extension)

**Reply by btaba** [reply]
Ah, the collisions are included automatically [here](https://github.com/google/brax/blob/main/brax/physics/colliders.py#L975) unless otherwise specified.

I believe you can add ` collide_include { }` to exclude all collisions

**Reply by Theo-Cheynel** [reply]
Damn, why didn't I think about this ! It works :wink: 

And for the record, here is the video with a working setup :
 

https://user-images.githubusercontent.com/34627497/212550934-387b4b68-62e9-4d7d-a421-a6bf6fdd00a4.mp4

**Reply by btaba** [reply]
great! glad it worked
